In [1]:
import pandas as pd
import numpy as np

## Revisión de unidades calamp 0330

In [2]:
ubd=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/ubd.csv',sep=';',encoding='latin1')
ubd['Unit Type'].fillna('Empty',inplace=True)
calamp_0330=ubd[ubd['Unit Type'].str.contains('0330')]
calamp_0330_filtered=calamp_0330[['Unit Type','Unit ID','Unit','Last Event Date','IMEI']]
calamp_0330_filtered_updated=calamp_0330_filtered[calamp_0330_filtered['Last Event Date'].str.contains('6/26/2025')]
imeis_0_500=calamp_0330_filtered_updated[0:500]
imeis_0_500['IMEI'].to_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/imeis_0_500.csv')
imeis_500_1000=calamp_0330_filtered_updated[500:1000]
imeis_500_1000['IMEI'].to_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/imeis_500_1000.csv')
imeis_1000_1327=calamp_0330_filtered_updated[1000:1327]
imeis_1000_1327['IMEI'].to_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/imeis_1000_1327.csv')


/tmp/ipykernel_957/1303271788.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  ubd['Unit Type'].fillna('Empty',inplace=True)


In [3]:
calamp_0330_ctc=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/devices-06-26-2025.csv')
calamp_0330_ctc.info()
calamp_0330_ctc_filtered=calamp_0330_ctc[['Model','Firmware Version','Configuration','Script Identifier','IMEI']]
calamp_0330_ctc_filtered.head(5)
for i in range(0,100):
    if calamp_0330_ctc_filtered.iloc[i,3]==7.72:
        print(calamp_0330_ctc_filtered.loc[i])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 64 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   ID                                    100 non-null    int64  
 1   Name                                  0 non-null      float64
 2   Identifier                            100 non-null    int64  
 3   Model                                 100 non-null    object 
 4   User ESN Mobile Id                    0 non-null      float64
 5   EOL                                   100 non-null    object 
 6   Enabled Services                      0 non-null      float64
 7   Asset                                 0 non-null      float64
 8   Account                               100 non-null    object 
 9   Status                                100 non-null    object 
 10  Last Event Time                       0 non-null      float64
 11  JPOD Firmware       

Las unidades que tienen problemas con el odómetro están asociadas al script 7.72 y el firmware 29b. Teniendo en cuenta que hay unidades con firmware 29b trabajando con otros script y no tienen problema, se concluye que el script 7.72 tiene algún inconveniente y consiste en que este script no está diseñado para unidades Calamp 0330.

Se procede a buscar todas las unidades calamp 0330 con script 7.72.

In [4]:
calamp_0330_ctc_1=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/devices_ctc_1.csv')
calamp_0330_ctc_2=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/devices_ctc_2.csv')
calamp_0330_ctc_3=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/devices_ctc_3.csv')
calamp_ctc_total=pd.concat([calamp_0330_ctc_1,calamp_0330_ctc_2,calamp_0330_ctc_3])
#calamp_ctc_total.info()
calamp_0330_verified=calamp_ctc_total[calamp_ctc_total['Model'].str.contains('330')]
calamp_0330_verified_filtered=calamp_0330_verified[['Model','Firmware Version','Script Identifier','IMEI']]
calamp_0330_verified_filtered.info()

for i in range(0,800):
    if calamp_0330_verified_filtered.iloc[i,2]==7.72:       
        #print(calamp_0330_verified_filtered.iloc[i,2])
        print(calamp_0330_verified_filtered.iloc[i,3])



<class 'pandas.core.frame.DataFrame'>
Index: 1083 entries, 0 to 250
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Model              1083 non-null   object 
 1   Firmware Version   1083 non-null   object 
 2   Script Identifier  1083 non-null   float64
 3   IMEI               1083 non-null   int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 42.3+ KB
358077092414644
359369090616414
358077091650081
359369090075629
352556102759268
352556102808172
352556102769622
359369090338803
359369090563624


In [161]:
proing_units=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/proing.csv',sep=';',encoding='latin1')
proing_0330=proing_units[proing_units['Unit Type'].str.contains('330')]
proing_0330_reporting_2025=proing_0330[proing_0330['Last Event Date'].str.contains('/2025')]
proing_0330_reporting_2025_06=proing_0330_reporting_2025[proing_0330_reporting_2025['Last Event Date'].str.contains('06/')]
#proing_0330_reporting_2025_06.info()
proing_0330_reporting_2025_06_filtered=proing_0330_reporting_2025_06[['Unit Type','IMEI']]
proing_0330_reporting_2025_06_filtered.info()
proing_0330_reporting_2025_06_filtered['IMEI'].to_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/proing_imeis.csv')
ctc_proing=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/ctc_proing.csv')
ctc_proing_filtered=ctc_proing[['Model','Firmware Version','Script Identifier','IMEI']]
ctc_proing_filtered.info()
for i in range(0,245):
    if ctc_proing_filtered.iloc[i,2]==7.72:       
        print(ctc_proing_filtered.iloc[i,2])
        print(ctc_proing_filtered.iloc[i,3])

<class 'pandas.core.frame.DataFrame'>
Index: 261 entries, 2 to 732
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Unit Type  261 non-null    object
 1   IMEI       261 non-null    object
dtypes: object(2)
memory usage: 6.1+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 245 entries, 0 to 244
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Model              245 non-null    object 
 1   Firmware Version   245 non-null    object 
 2   Script Identifier  245 non-null    float64
 3   IMEI               245 non-null    int64  
dtypes: float64(1), int64(1), object(2)
memory usage: 7.8+ KB
7.72
352556102769622


Se crea tarea para cargar script 7.163 a las unidades.
358077092414644,9101161085
359369090616414,9103424953
358077091650081,9103403704
359369090075629,9103409944
352556102759268,9103391572
352556102808172,9101159104
352556102769622,9101431623
359369090338803,9103434812
359369090563624,9103382751

## FJO660

IMEI: 352556102769622

Serial: 1363187479

Firmware: 29f

Script: 7.72

Trama de 7 acumuladores: 0x830513631874790101010A1610685C7198029E8170D336EB6C004104050F0205080400FB6100003264000000000000000000000000000000000000000000000000

Se cambió a Basic Accum 25-06-2025.

Se cambió a Basic Accum with rpm, odometer &.... 26-06-2025 8:34am

In [ ]:
basic_report_fjo660=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/fjo660.csv')
basic_report_fjo660.info()
miles_driven_fjo660=basic_report_fjo660[['MilesDriven','textbox43','StartTime']]
miles_driven_fjo660
#drive_events=basic_report_fjo660[basic_report_fjo660['textbox43'].str.contains('Drive')]
#drive_events.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   GroupName         0 non-null      float64
 1   UnitName          175 non-null    object 
 2   StartTime         175 non-null    object 
 3   MovingTimeSecs    175 non-null    object 
 4   MilesDriven       175 non-null    object 
 5   MaxSpeed          175 non-null    int64  
 6   StopTime          175 non-null    object 
 7   textbox43         175 non-null    object 
 8   StopLengthSecs    175 non-null    object 
 9   ContactName       175 non-null    object 
 10  textbox45         175 non-null    object 
 11  textbox54         175 non-null    int64  
 12  textbox28         175 non-null    object 
 13  textbox29         175 non-null    object 
 14  textbox30         175 non-null    int64  
 15  textbox32         175 non-null    object 
 16  MovingTimeSecs_1  175 non-null    object 
 1

MilesDriven    textbox43             StartTime
0          0,00  Parked/Idle  9/06/2025 12:00 a.m.
1          0,00  Parked/Idle   9/06/2025 6:21 a.m.
2          0,00         Idle   9/06/2025 6:23 a.m.
3          0,00         Idle   9/06/2025 6:41 a.m.
4          0,00         Idle   9/06/2025 6:45 a.m.
..          ...          ...                   ...
170        0,00      Ign Off  13/06/2025 3:56 p.m.
171        0,00  Parked/Idle  13/06/2025 4:06 p.m.
172        0,00  Parked/Idle  13/06/2025 4:41 p.m.
173        0,00         Idle  13/06/2025 4:42 p.m.
174        0,00      Ign Off  13/06/2025 4:45 p.m.

[175 rows x 3 columns]

In [3]:
hl_fjo660=pd.read_csv('/home/jlhb1984/py-projects/data/calamp0330 troubles/historicallocations.csv',sep=';',encoding='latin1')
hl_fjo660.info()
hl_fjo660_drive_event=hl_fjo660[hl_fjo660['Events'].str.contains('Drive')]
hl_fjo660_drive_event.head(15)
#hl_fjo660_drive_event.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1418 entries, 0 to 1417
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     1418 non-null   int64  
 1   Unit         1418 non-null   object 
 2   Events       1418 non-null   object 
 3   Date         1418 non-null   object 
 4   Landmark     438 non-null    object 
 5   Address      1418 non-null   object 
 6   Speed        1418 non-null   int64  
 7   Temperature  0 non-null      float64
 8   Driver       0 non-null      float64
 9   Latitude     1418 non-null   float64
 10  Longitude    1418 non-null   float64
dtypes: float64(4), int64(2), object(5)
memory usage: 122.0+ KB


Sequence    Unit Events                  Date                 Landmark  \
64      1354  FJO660  Drive  Jun 13 2025 04:48 PM  BODEGA PRINCIPAL IBAGUE   
66      1352  FJO660  Drive  Jun 13 2025 04:47 PM                      NaN   
67      1351  FJO660  Drive  Jun 13 2025 04:45 PM                      NaN   
68      1350  FJO660  Drive  Jun 13 2025 04:45 PM                      NaN   
69      1349  FJO660  Drive  Jun 13 2025 04:45 PM                      NaN   
70      1348  FJO660  Drive  Jun 13 2025 04:45 PM                      NaN   
71      1347  FJO660  Drive  Jun 13 2025 04:43 PM                      NaN   
72      1346  FJO660  Drive  Jun 13 2025 04:42 PM                      NaN   
80      1338  FJO660  Drive  Jun 13 2025 03:56 PM                      NaN   
81      1337  FJO660  Drive  Jun 13 2025 03:56 PM                      NaN   
82      1336  FJO660  Drive  Jun 13 2025 03:54 PM                      NaN   
83      1335  FJO660  Drive  Jun 13 2025 03:53 PM                      NaN   
84      1334  FJO660  Drive  Jun 13 2025 03:53 PM                      NaN   
85      1333  FJO660  Drive  Jun 13 2025 03:53 PM                      NaN   
86      1332  FJO660  Drive  Jun 13 2025 03:51 PM                      NaN   

                                              Address  Speed  Temperature  \
64                                   Avenida Picaleña      0          NaN   
66  Km 10 de la vía Buenos Aires - Picalena, , Tolima     19          NaN   
67                              Av 40, Ibague, Tolima     28          NaN   
68                              Av 40, Ibague, Tolima     28          NaN   
69                              Av 40, Ibague, Tolima     27          NaN   
70                              Av 40, Ibague, Tolima     12          NaN   
71                     40TL, Ibagué, Tolima, Colombia      0          NaN   
72  Km 1 de la vía Picalena - Santa Rita, , Tolima...      9          NaN   
80  Km 1 de la vía Picalena - Santa Rita, , Tolima...      9          NaN   
81                     40TL, Ibagué, Tolima, Colombia      0          NaN   
82                              Av 40, Ibague, Tolima     16          NaN   
83                              Av 40, Ibague, Tolima      0          NaN   
84                    Av 40, Ibague, Tolima, Colombia     16          NaN   
85                    Av 40, Ibague, Tolima, Colombia     24          NaN   
86                              Av 40, Ibague, Tolima     41          NaN   

    Driver  Latitude  Longitude  
64     NaN  4.394681 -75.137272  
66     NaN  4.397397 -75.140513  
67     NaN  4.399654 -75.145551  
68     NaN  4.399699 -75.145838  
69     NaN  4.400021 -75.145995  
70     NaN  4.400316 -75.145744  
71     NaN  4.406195 -75.143897  
72     NaN  4.406586 -75.144722  
80     NaN  4.407665 -75.145933  
81     NaN  4.406536 -75.145688  
82     NaN  4.401358 -75.147965  
83     NaN  4.401009 -75.147926  
84     NaN  4.400893 -75.147930  
85     NaN  4.400830 -75.148066  
86     NaN  4.404310 -75.158592

## Los contadores del odómetro están bien:

0x830513631874790101010A0634684C9C6C029EFD53D33677B80078140A0F03030803F3FDF8000036AC000000000000000000000000000000000000000000000000 (Accum 0= 66321912)

0x830513631874790101010A0620684C8F5E02A00B5DD333B57D00542B0E0F03030803F3EEF7000036B0000000000000000000000000000000000000000000000000 (Accum 0=66318071)






In [4]:
hl_fjo660_drive_event=hl_fjo660[hl_fjo660['Events'].str.contains('Ignition')]
hl_fjo660_drive_event.head(5)

Sequence    Unit        Events                  Date  \
56      1362  FJO660  Ignition OFF  Jun 13 2025 04:54 PM   
74      1344  FJO660   Ignition ON  Jun 13 2025 04:41 PM   
76      1342  FJO660  Ignition OFF  Jun 13 2025 04:06 PM   
77      1341  FJO660   Ignition ON  Jun 13 2025 04:06 PM   
78      1340  FJO660  Ignition OFF  Jun 13 2025 03:57 PM   

                   Landmark  \
56  BODEGA PRINCIPAL IBAGUE   
74                      NaN   
76                      NaN   
77                      NaN   
78                      NaN   

                                              Address  Speed  Temperature  \
56                                   Avenida Picaleña      0          NaN   
74     Unnamed Road, Ibagué, Ibagué, Tolima, Colombia      0          NaN   
76     Unnamed Road, Ibagué, Ibagué, Tolima, Colombia      0          NaN   
77     Unnamed Road, Ibagué, Ibagué, Tolima, Colombia      0          NaN   
78  Km 1 de la vía Picalena - Santa Rita, , Tolima...      0          NaN   

    Driver  Latitude  Longitude  
56     NaN  4.394449 -75.137329  
74     NaN  4.407834 -75.145578  
76     NaN  4.407853 -75.145590  
77     NaN  4.407850 -75.145589  
78     NaN  4.407852 -75.145602

El 27-06-2025 la unidad FJO660 tomó el script 7.163, pero no ha mostrado evento Drive para verificar que el problema fue solucionado.